src/main/java/org/fountainmc/forge/mixin/entity/MixinEntity.java

```diff
@@ -24,32 +24,151 @@
 
 package org.fountainmc.forge.mixin.entity;
 
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import net.minecraft.command.ICommandSender;
 import net.minecraft.util.math.BlockPos;
 import org.fountainmc.api.entity.Entity;
 import org.fountainmc.api.world.Location;
 import org.fountainmc.api.world.World;
-import org.spongepowered.asm.mixin.Mixin;
-import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.*;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
 
 @Mixin(net.minecraft.entity.Entity.class)
-public abstract class MixinEntity implements Entity {
+@Implements(@Interface(iface = Entity.class, prefix = "entity$"))
+public abstract class MixinEntity implements Entity, ICommandSender {
 
     @Shadow public boolean onGround;
+    @Shadow public float rotationPitch;
+    @Shadow public float rotationYaw;
 
     @Shadow
     public abstract BlockPos getPosition();
 
     @Shadow
     public abstract net.minecraft.world.World getEntityWorld();
 
+    @Shadow
+    public abstract void setPosition(double x, double y, double z);
+
+    @Shadow
+    public abstract List<net.minecraft.entity.Entity> shadow$getPassengers();
+
+    @Shadow
+    public abstract void shadow$addPassenger(net.minecraft.entity.Entity passenger);
+
+    @Shadow
+    public abstract void removePassenger(net.minecraft.entity.Entity passenger);
+
     @Override
     public Location getLocation() {
         return new Location((World) this.getEntityWorld(),
                 this.getPosition().getX(), this.getPosition().getY(), this.getPosition().getZ());
     }
 
     @Override
+    public void teleport(Location destination) {
+        this.setPosition(destination.getX(), destination.getY(),
+                destination.getZ());
+    }
+
+    @Override
+    public float getPitch() {
+        return this.rotationPitch;
+    }
+
+    @Override
+    public void setPitch(float pitch) {
+        this.rotationPitch = pitch;
+    }
+
+    @Override
+    public float getYaw() {
+        return this.rotationYaw;
+    }
+
+    @Override
+    public void setYaw(float yaw) {
+        this.rotationYaw = yaw;
+    }
+
+    @Override
     public boolean isOnGround() {
         return this.onGround;
     }
+
+    @Nullable
+    @Override
+    public Entity getPrimaryPassenger() {
+        if (getPassengers().size() != 0) {
+            return getPassengers().get(0);
+        }
+        return null;
+    }
+
+    @Intrinsic
+    @Nonnull
+    public ImmutableList<Entity> entity$getPassengers() {
+        List entities = this.shadow$getPassengers();
```

Comment:
It doesn't work unless the shadow method has a List return type without a generic. Would Mixin register it correctly?